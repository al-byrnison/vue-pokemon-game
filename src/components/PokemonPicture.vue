<template>
	<div class="pokemon-container">
		<img
			:src="imgSrc"
			alt="pokemon"
			class="pokemon-hidden"
		/>
		<img
			v-if="showPokemon"
			:src="imgSrc"
			alt="pokemon"
			class="fade-in"
		/>
	</div>
</template>

<script>
	export default {
		props: {
			pokemonId: {
				type: Number,
				required: true,
			},
			showPokemon: {
				type: Boolean,
				required: true,
				default: false,
			},
		},
		computed: {
			imgSrc() {
				return `https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/${this.pokemonId}.svg`
			},
		},
	}
</script>

<style scoped>
	/* Pokemon Picture */
	.pokemon-container {
		position: relative;
		height: 12rem;
	}
	img {
		height: 12rem;
    width: 12rem;
		position: absolute;
		top: 50%;
		left: 50%;
		transform: translate(-50%, -50%);
		user-select: none;
		-moz-user-select: none;
		-ms-user-select: none;
		-webkit-user-drag: none;
		-webkit-user-select: none;
	}
	.pokemon-hidden {
		filter: grayscale(100%) brightness(0%);
	}
</style>
