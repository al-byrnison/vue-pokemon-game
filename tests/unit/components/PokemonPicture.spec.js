import { shallowMount } from '@vue/test-utils'
import PokemonPicture from '@/components/PokemonPicture'

describe('PokemonPicture component', () => {
	test('should snapshot match', () => {
		const wrapper = shallowMount(PokemonPicture, {
			props: {
				pokemonId: 1,
				showPokemon: true,
			},
		})
		expect(wrapper.html()).toMatchSnapshot()
	})

	test('should show hidden img pokemon 100', () => {
		const wrapper = shallowMount(PokemonPicture, {
			props: {
				pokemonId: 100,
				showPokemon: false,
			},
		})
		const [img1, img2] = wrapper.findAll('img')
		expect(img1.exists()).toBeTruthy()
		expect(img2).toBe(undefined)
		expect(img1.classes('pokemon-hidden')).toBeTruthy()
        
        const srcImg = 'https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/100.svg'
        expect(img1.attributes('src')).toBe(srcImg)
	})

	test('should show pokemon if showPokemon:true', () => {
		const wrapper = shallowMount(PokemonPicture, {
			props: {
				pokemonId: 100,
				showPokemon: true,
			},
		})
		const imgPokemon = wrapper.find('img')
		expect(imgPokemon.exists()).toBeTruthy()

		expect(imgPokemon.classes('pokemon-hidden')).toBeFalsy()
        expect(imgPokemon.classes('fade-in')).toBeTruthy()
	})
})
