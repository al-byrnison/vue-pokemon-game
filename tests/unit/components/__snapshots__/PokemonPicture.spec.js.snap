// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonPicture component should snapshot match 1`] = `<div class="pokemon-container"><img src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/other/dream-world/1.svg" alt="pokemon" class="fade-in"></div>`;
